st2actions/st2actions/container/base.py

```diff
@@ -127,6 +128,40 @@ def _do_run(self, runner):
                     (status, result, context) = runner.run(action_params)
                     result = jsonify.try_loads(result)
 
+            LOG.debug('Validating runner output: %s', runner.runner_type.output_schema)
+
+            if runner.runner_type.output_schema.get('unmodeled'):
+                runner.runner_type.output_schema.pop('unmodeled')
+                LOG.warn(
+                    """Deprecation Notice: This runner has previously had unmodeled
+                    output. In StackStorm 3.1 the output will be placed under the
+                    `output` key."""
+                )
+                runner_schema = {
+                    "additionalProperties": True
+                }
+            else:
+                runner_schema = {
+                    "type": "object",
+                    "properties": runner.runner_type.output_schema,
+                    "additionalProperties": False
+                }
+
+            LOG.info("Action Result: %s", result)
+
+            schema.validate(result, runner_schema, cls=schema.get_validator('custom'))
+
+            if runner.action.output_schema:
+                output_schema = copy.deepcopy(runner.runner_type.output_schema)
```

Comment:
Maybe now that we have `fast_deepcopy` function it might make sense to use that. Or perhaps since it's a small dict it's not a big deal.

Reply:
I'll use it for consistency now that it exists. Plus never know. People do crazy things sometimes. That 1000 param output action will exist one day.